src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -732,6 +735,13 @@ private void visitType(JSType typeToVisit) {
       // See also JsdocToEs6TypedConverter in the Closure code base. This code is implementing the
       // same algorithm starting from JSType nodes (as opposed to JSDocInfo), and directly
       // generating textual output. Otherwise both algorithms should produce the same output.
+      if (isPrivate(typeToVisit.getJSDocInfo())) {
+        // TypeScript does not allow public APIs that expose non-exported/private types. Just emit
+        // an empty object literal type for those, i.e. something that cannot be used for anything,
+        // except being passed around.
+        emit("{}");
```

Comment:
How about adding `interface PrivateType {}` to closure.lib.d.ts and used here, so when this errors out the users will get a more descriptive message. Typewise, AFAIKT `{}` is equivalent to empty interface.